docs/user/features/codeql.md

```diff
@@ -0,0 +1,110 @@
+4. In addition to other features described in the [CodeQL overview](https://codeql.github.com/docs/codeql-overview/)
+
+## CodeQL Usage
+
+CodeQL provides the capability to debug the actual queries and for our (Tianocore) community to write our own queries
+and even contribute back to the upstream repo when appropriate. In other cases, we might choose to keep our own queries
+in a separate TianoCore repo or within a directory in the edk2 code tree.
+
+This is all part of CodeQL Scanning. [This page](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning)
+has information concerning how to configure CodeQL scanning within a GitHub project such as edk2. Information on the
+topic of running additional custom queries is documented [here](https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/configuring-code-scanning#running-additional-queries)
+in that page.
+
+In addition, CodeQL offers the flexibility to:
+
+- Build databases locally
+- Retrieve databases from server builds
+- Relatively quickly test queries locally against a database for a fast feedback loop
+- Suppress false positives
+- Customize the files and queries used in the edk2 project and quickly keep this list in sync between the server and
+  local execution
+
+## EDK2 PyTool Extensions CodeQL Module
+
+A Python module located in `edk2toolext/codeql.py` provides helper functions that platform and CI build files can use
+to support CodeQL enabled builds with minimal effort.
+
+The following functions are available. The expected actions to be taken per scope are included as well.
+
+- `add_command_line_option()` - Adds the CodeQL command (`--codeql`) to the platform command line options.
+- `get_scopes()` - Returns the active CodeQL scopes for this build.
+  - Host OS neutral scopes:
+    - `codeql-build` - Build the CodeQL database during firmware build.
+    - `codeql-analyze` - Analyze the CodeQL database after firmware build (post-build).
+  - Linux scope:
+    - `codeql-linux-ext-dep` - Download the Linux CodeQL CLI external dependency.
+  - Windows scope:
+    - `codeql-windows-ext-dep` - Download the Windows CodeQL CLI external dependency.
+- `is_codeql_enabled_on_command_line` - Returns whether CodeQL was enabled (via `--codeql`) on the command line.
+- `set_audit_only_mode` - Configures the CodeQL plugin to run in audit only mode.
+
+These functions are intended to ease repo integration with CodeQL, ensure consistent command-line usage across repos,
+and define standard scopes that other plugins and tools can depend on for CodeQL operations.
+
+### Integration Examples
+
+This section provides examples of how to use the functions available in `edk2toolext/codeql.py`.
+
+#### add_common_line_option()
+
+Call when the command-line options are registered for the build:
+
+```python
+    import edk2toolext.codeql as codeql
```

Comment:
Not sure if you meant to use this as `import edk2toolext.codeql as codeql_helpers`
If you didn't, the as is not necessary.
```suggestion
    import edk2toolext.codeql
```